src/modules/rlm_pap/rlm_pap.c

```diff
@@ -375,11 +401,19 @@ static rlm_rcode_t CC_HINT(nonnull) mod_authorize(void *instance, REQUEST *reque
 
 #ifdef HAVE_OPENSSL_EVP_H
 		case PW_SHA2_PASSWORD:
+		case PW_SSHA224_PASSWORD:
+		case PW_SSHA256_PASSWORD:
+		case PW_SSHA384_PASSWORD:
+		case PW_SSHA512_PASSWORD:
```

Comment:
It would seem that the original SSHA1 code did not place delimiters in the encoding of the hash (I expect as there was one size), and the SHA2 code did not place a length in the string as it was redundant. Unfortunately if you has say an SSHA256 with a 32-byte salt, it would seem to be the size of a SSHA512.